bor/README.md

```diff
@@ -0,0 +1,166 @@
+where ,
+
+* `ID` means the id of the span, calculated by monotonically incrementing the ID of the previous span.
+* `StartBlock` corresponds to the block from which the given span would commence.
+* `EndBlock` corresponds to the block at which the given span would conclude.
+* `ValidatorSet` defines the set of active validators.
+* `SelectedProducers` are the validators selected to produce blocks from the validator set.
+* `ChainID` corresponds to bor chain ID.
+
+A validator on heimdall can construct a span proposal message:
+
+```
+type MsgProposeSpan struct {
+    ID         uint64                  `json:"span_id"`
+    Proposer   hmTypes.HeimdallAddress `json:"proposer"`
+    StartBlock uint64                  `json:"start_block"`
+    EndBlock   uint64                  `json:"end_block"`
+    ChainID    string                  `json:"bor_chain_id"`
+    Seed       common.Hash             `json:"seed"`
+}
+```
+
+Upon broadcasting the message, it is initially checked by `HandleMsgProposeSpan` handler for basic sanity (verify whether the proposed span is in continuity, appropriate span duration, correct chain ID, etc.). Since this is a side-transaction, the validators then vote on the data present in `MsgProposeSpan` on the basis of its correctness. All these checks are done in `SideHandleMsgSpan` (verifying `Seed`, span continuity, etc) and if correct, the validator would vote `YES`.
+Finally, if there are 2/3+ `YES` votes, the `PostHandleMsgEventSpan` persists the proposed span in the state via the keeper :  
+
+```
+err := k.FreezeSet(ctx, msg.ID, msg.StartBlock, msg.EndBlock, msg.ChainID, msg.Seed)
+if err != nil {
+	k.Logger(ctx).Error("Unable to freeze validator set for span", "Error", err)
+	return common.ErrUnableToFreezeValSet(k.Codespace()).Result()
+}
+```
+
+`FreezeSet` internally invokes `SelectNextProducers`, which pseudo-randomly picks producers from the validaor set, leaning more towards validators with higher stake:
+
+```
+// select next producers
+newProducers, err := k.SelectNextProducers(ctx, seed)
+if err != nil {
+	return err
+}
+```
+
+and then intialises and stores the span:
+
+```
+newSpan := hmTypes.NewSpan(
+	id,
+	startBlock,
+	endBlock,
+	k.sk.GetValidatorSet(ctx),
+	newProducers,
+	borChainID,
+)
+
+return k.AddNewSpan(ctx, newSpan)
+```
+
+## HOW TO PROPOSE A SPAN ?
+
+A validator can leverage the CLI to propose a span like so :
+
+```
+heimdallcli tx bor propose-span --proposer <VALIDATOR ADDRESS> --start-block <START_BLOCK> --span-id <SPAN_ID> --bor-chain-id <BOR_CHAIN_ID>
+```
+
+Or the rest server : 
+
+```
+curl -X POST "localhost:1317/bor/propose-span?bor-chain-id=<BOR_CHAIN_ID>&start-block=<START_BLOCK>&span-id=<SPAN_ID>"
+```
+
+## QUERY COMMANDS
+
+One can run the following query commands from the bor module :
+
+* `span` - Query the span corresponding to the given span id:
+
+via CLI
+```
+heimdallcli query bor span --span-id=<span ID here>
+```
+
+via Rest
+```
+curl localhost:1317/bor/span/1000
```

Comment:
Span id is hardcoded here. Is it intentional?

Reply:
Rectified [4fc537d](https://github.com/maticnetwork/heimdall/pull/1071/commits/4fc537d495cb854277cacff52c1b4a6282ff0bdf).